.github/workflows/ossf_scorecard.yml

```diff
@@ -35,11 +35,12 @@ jobs:
           repo_token: ${{ secrets.SCORECARD_TOKEN }}  # read-only fine-grained token to read branch protection settings
 
       - name: "Upload results"
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
         with:
           name: SARIF file
           path: results.sarif
           retention-days: 5
+          overwrite: true
```

Comment:
@dreamorosi is that all you needed to do?

I read the migration guide and it looks alright, since we're using a static artifact name.

Reply:
Yep, that's all, but in a different place.

My understanding is that if the workflow(s) don't mutate the artifact in the same run, you don't need to do any change. At least for us, this is the case for the OSSF scorecard workflow ([source](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/workflows/ossf_scorecard.yml)).

In our repo we do overwrite the artifact in the reusable publish layer workflow ([source](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/workflows/reusable_deploy_layer_stack.yml#L100)), since we run that in a matrix we had to add the `overwrite` key.

I'm not 100% familiar with that part of your workflows, but I think you should check if it needs to be added [here](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/workflows/reusable_deploy_v2_layer_stack.yml#L205). For us we found the issue during a release, but since we have [a canary step](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/workflows/publish_layer.yml#L54-L65), it caused no disruption.